modular_skyrat/modules/cryosleep/code/cryopod.dm

```diff
@@ -304,8 +306,14 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 	else
 		crew_member["job"] = "N/A"
 
-	// Delete them from datacore.
+	// Delete them from datacore and ghost records.
 	var/announce_rank = null
+
+	for(var/datum/data/record/record as anything in GLOB.ghost_records)
+		if(record.fields["name"] == mob_occupant.real_name)
+			announce_rank = record.fields["rank"]
+			qdel(record)
```

Comment:
This way you won't have nulls in that list. (I think that's how the remove proc is written).
```suggestion
			announce_rank = record.fields["rank"]
			GLOB.ghost_records.Remove(record)
			qdel(record)
```